x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleShardTaskParams.java

```diff
@@ -108,7 +110,9 @@ public void writeTo(StreamOutput out) throws IOException {
         shardId.writeTo(out);
         out.writeStringArray(metrics);
         out.writeStringArray(labels);
-        out.writeOptionalStringArray(dimensions);
+        if (out.getTransportVersion().onOrAfter(TransportVersions.DIMENSIONS_SERIALIZATION_PERSISTENT_TASK)) {
+            out.writeOptionalStringArray(dimensions);
+        }
```

Comment:
Do we need to add a `if (dimensions.length != 0) {` check in the `toXContent(...)` method of this class? Just wondering if the xcontent format is shared between nodes?

Reply:
++, added a serializing test for this in https://github.com/elastic/elasticsearch/pull/106878/commits/7593cd8a410962cd64c7a0af0a2255708fc3dc56.